### ISSUE TITLE
test_list_datasets_with_high_size_parameter  fails often

### ISSUE THREAD

**Issue by janvanrijn** [openml/openml-python#626]
This function seems to fail often. 
```
    def test_list_datasets_with_high_size_parameter(self):
        datasets_a = openml.datasets.list_datasets()
        datasets_b = openml.datasets.list_datasets(size=np.inf)
    
>       self.assertEqual(len(datasets_a), len(datasets_b))
E       AssertionError: 1505 != 1506
```

My assumption: due to multi threaded testing, in between the two list dataset calls some new datasets have been uploaded. 

We should redesign this test case.

**Comment by janvanrijn** [comment]
Solution in https://github.com/openml/openml-python/pull/628

**Comment by mfeurer** [comment]
Closed via #628.
